src/Feature/Terminal/Feature_Terminal.re

```diff
@@ -71,6 +72,35 @@ type outmsg =
 
 let shellCmd = ShellUtility.getDefaultShell();
 
+// CONFIGURATION
+
+module Configuration = {
+  open Oni_Core;
+  open Config.Schema;
+
+  let shellCommandWindows =
+    setting("terminal.integrated.shell.windows", string, ~default=shellCmd);
+  let shellCommandLinux =
+    setting("terminal.integrated.shell.linux", string, ~default=shellCmd);
+  let shellCommandOSX =
+    setting("terminal.integrated.shell.osx", string, ~default=shellCmd);
+
+  let shellArgsWindows =
+    setting(
+      "terminal.integrated.shellArgs.windows",
+      list(string),
+      ~default=[],
+    );
+  let shellArgsLinux =
+    setting(
+      "terminal.integrated.shellArgs.linux",
+      list(string),
+      ~default=[],
+    );
+  let shellArgsOSX =
+    setting("terminal.integrated.shellArgs.osx", list(string), ~default=[]);
```

Comment:
The convention I've used elsewhere is to make every dot-separated segment except the last a module, as that makes the translation to and from the keys really easy and predictable. Assuming it would be implied that everything here is prefixed with `terminal.integrated` you could for example have `Configuration.Shell.windows` and `Configuration.ShellArgs.osx`.

Reply:
Sounds good!